src/components/HomePage.jsx

```diff
@@ -0,0 +1,121 @@
+import './Styles/HomePage.css';
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import Countries from './Countries';
+import { getCountriesData } from '../Redux/Countries/CountriesSlice';
+import world from '../images/World.svg';
+import Africa from '../images/Africa.svg';
+import Americas from '../images/Americas.svg';
+import Asia from '../images/Asia.svg';
+import Europe from '../images/Europe.svg';
+import Oceania from '../images/Oceania.svg';
+
+const HomePage = () => {
+  const { isLoading, error, countries } = useSelector(
+    (state) => state.countries,
+  );
+  const [countriesData, setCountriesData] = useState(countries);
+  const [selected, setSelected] = useState();
+  const [map, setMap] = useState(world);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (!isLoading) {
+      dispatch(getCountriesData());
+      setCountriesData(null);
+    }
+  }, [dispatch, isLoading]);
+
+  const filterCountries = (e) => {
+    setSelected(e.target.value);
+    if (e.target.value === 'All') {
+      setCountriesData(countries);
+      setMap(world);
+    } else {
+      setCountriesData(
+        countries.filter((country) => country.region === e.target.value),
+      );
+      switch (e.target.value) {
+        case 'Asia':
+          setMap(Asia);
+          return;
+        case 'Europe':
+          setMap(Europe);
+          return;
+        case 'Africa':
+          setMap(Africa);
+          return;
+        case 'Americas':
+          setMap(Americas);
+          return;
+        case 'Oceania':
+          setMap(Oceania);
+          return;
+        default:
+          setMap(world);
+      }
+    }
+  };
+  const filterCountries2 = (e) => {
+    setCountriesData(
+      // eslint-disable-next-line max-len
```

Comment:
- [ ] Please always make sure that you are not disabling the `linters` either *intentionally* or **unintentionally** because by doing this you are compromising the `quality` of the code, which as a developer is not a good practice and can cause problems, which will be difficult to debug at the later stages of development.